<!doctype html>
<html lang="en">

	<head>
		<meta charset="utf-8">

		<title>Manipulation de données et production de figures sous R</title>

		<meta name="description" content="dplyR, tidyR et ggplot2">
		<meta name="author" content="collectif de bioinformatique de Baillarguet">

		<meta name="apple-mobile-web-app-capable" content="yes">
		<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">

		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">

		<link rel="stylesheet" href="css/reveal.css">
		<link rel="stylesheet" href="css/theme/solarized.css" id="theme">

		<!-- Code syntax highlighting -->
		<link rel="stylesheet" href="lib/css/zenburn.css">

		<!-- Printing and PDF exports -->
		<script>
			var link = document.createElement( 'link' );
			link.rel = 'stylesheet';
			link.type = 'text/css';
			link.href = window.location.search.match( /print-pdf/gi ) ? 'css/print/pdf.css' : 'css/print/paper.css';
			document.getElementsByTagName( 'head' )[0].appendChild( link );
		</script>
		
		
		<script src="lib/htmlwidgets-0.5/htmlwidgets.js"></script>
		<script src="lib/plotlyjs-1.1.1/plotly-latest.min.js"></script>
		<script src="lib/plotly-binding-2.0.6/plotly.js"></script>

		
		<!--[if lt IE 9]>
		<script src="lib/js/html5shiv.js"></script>
		<![endif]-->
	</head>

	<body>

	  <div class="reveal">
		<div class="slides">
          
          <!-- Titre -->
		  <section>
			<h1>Manipulation de données et visualisation sous R</h1>
			<h3>ggplot2, dplyr & tidyr</h3>
			<p>
			  <small>collectif de bioinformatique de Baillarguet</small>
			</p>
		  </section>

		  <!-- Historique de R -->
		  <section>
			<h2>R</h2>
			<a href="https://www.r-project.org/">
              <img data-src="https://www.r-project.org/Rlogo.png" alt="R logo">
            </a>
            <ul>
			  <li>1993</li>
			  <li>logiciel libre</li>
			  <li>multi-plateforme</li>
              <li>modulaire et en forte croissance</li>
			</ul>
		  </section>

		  <!-- Popularité de R -->
		  <section>
			<h2>Popularité croissante</h2>
			<a href="http://r4stats.com/articles/popularity/">
              <img data-src="http://i1.wp.com/datasciencepopularity.com/wp-content/uploads/2015/05/fig_2f_scholarlyimpactallstat21.png" alt="R's popularity">
            </a>
		  </section>

		  <!-- Nombre de modules -->
		  <section>
			<h2>Nombre de modules croissant</h2>
			<a href="http://r4stats.com/articles/popularity/">
              <img
                 width="500"
                 data-src="http://i1.wp.com/r4stats.com/wp-content/uploads/2015/05/fig_9_cran.png"
                 alt="R's modules">
            </a>
		  </section>

		  <!-- Hadley Wickham -->
		  <section>
			<h2>Hadley Wickham</h2>
            <a href="https://www.youtube.com/watch?v=LOXe6Eu59As">
              <img
                 data-src="https://i.ytimg.com/vi/LOXe6Eu59As/maxresdefault.jpg"
                 alt="Hadley Wickham">
            </a>
		  </section>

		  <!-- Hadley Wickham modules -->
		  <section>
			<h2>Hadley Wickham</h2>
			<h3>manipulation (unifiée) et visualisation de données</h3>
            <ul>
			  <li><a href="https://cran.r-project.org/web/packages/stringr/index.html">stringr</a></li>
			  <li><a href="https://cran.r-project.org/web/packages/dplyr/index.html">dplyr</a></li>
			  <li><a href="https://cran.r-project.org/web/packages/tidyr/index.html">tidyr</a></li>
			  <li><a href="http://ggplot2.org/">ggplot2</a> (<a href="https://www.springer.com/us/book/9780387245447">grammar of graphics</a> par Leland Wilkinson)</li>
			  <li><a href="http://ggvis.rstudio.com/">ggvis</a></li>
			  <li>Chief Scientist <a href="https://www.rstudio.com/">RStudio</a></li>
			</ul>
		  </section>

		  <!-- RStudio -->
		  <section>
			<h2>RStudio</h2>
			<h3>interface graphique pour R (et bien plus)</h3>
            <a href="https://www.rstudio.com/">
              <img width="400"
                   data-src="https://www.rstudio.com/wp-content/uploads/2014/06/RStudio-Ball.png"
                   >
            </a>
		  </section>

		  <!-- Première démonstration : données simples -->
		  <section>
			<h2>Cleaning and visualizing genomic data</h2>
			<h3>a case study in tidy analysis</h3>
            <a href="http://varianceexplained.org/r/tidy-genomics/">
              <img width="200"
                   data-src="http://varianceexplained.org/images/david_robinson_picture2.jpg"
                   >
            </a>
            <p>par David Robinson</p>
		  </section>

		  <!-- Source des données -->
		  <section>
			<h2>Coordination of Growth Rate, Cell Cycle, Stress Response, and Metabolic Activity in Yeast</h2>
			<h3>Brauer et al. (2008) Mol. Biol. Cell January 1, 2008 vol. 19 no. 1 352-367 (doi: 10.1091/mbc.E07-08-0779)</h3>
            <p><a href="http://varianceexplained.org/files/Brauer2008_DataSet1.tds">http://varianceexplained.org/files/Brauer2008_DataSet1.tds</a></p>
		  </section>

		  <!-- Premières commandes avec RStudio -->
		  <section>
			<h2>Premières commandes avec RStudio</h2>
			<pre>
              <code class="R" data-trim contenteditable>
Brauer2008_DataSet1 <- read.delim("/tmp/Rtmp3mlo42/data1aaa3ddc8686")
View(Brauer2008_DataSet1)                
dim(Brauer2008_DataSet1)
			  </code>
            </pre>
			<p>Nombre de colonnes et de lignes</p>
		  </section>

		  <!-- Description des données -->
		  <section>
			<h2>Description des données</h2>
            <p>6 milieux de culture en déficit d'un nutriment :</p>
            <ul>
			  <li>glucose (G)</li>
			  <li>ammonium (N)</li>
			  <li>phosphate (P)</li>
			  <li>sulfate (S)</li>
			  <li>leucine (L)</li>
			  <li>uracil (U)</li>
			</ul>
            <p>6 vitesses de croissance: lent (0,05) à rapide (0,3).</p>
		  </section>

		  <!-- Pourquoi changer la disposition des données ? -->
		  <section>
			<h2>Disposition des données : pourquoi changer ?</h2>
            <p>Les outils d'Hadley Wickham réclament :</p>
            <ul>
			  <li>une seule variable par colonne,</li>
			  <li>une seule observation par ligne,</li>
			  <li>un seul type d'observation par tableau.</li>
			</ul>
            <p>Ce qui ne va pas dans la disposition actuelle :</p>
            <ul>
			  <li class="fragment">G0.05 n'est pas un nom de variable,</li>
			  <li class="fragment">plusieurs variables dans la colonne NAME.</li>
			</ul>
		  </section>

		  <!-- Traitement de la colonne NAME ? -->
		  <section>
			<h2>Séparer une colonne en plusieurs variables</h2>
            <p>La colonne NAME contient :</p>
            <ul>
			  <li>un nom de gène (ex : SFB2) ou rien,</li>
			  <li>un nom de processus biologique,</li>
			  <li>une fonction moléculaire,</li>
			  <li>un identifiant,</li>
			  <li>un autre identifiant (?)</li>
			</ul>
            <p>Séparer les variables facilite l'exploration et permet de poser des questions plus précises.</p>
		  </section>

		  <!-- Utilisation de dplyr et tidyr -->
		  <section>
		    <section>
			  <h2>Séparer une colonne en plusieurs variables</h2>
              <p>fonction <i>separate</i> de tidyr :</p>
			  <pre>
                <code class="R" data-trim>
library(dplyr)
library(tidyr)
cleaned_data <- Brauer2008_DataSet1 %>%
    separate(NAME, c("name", "BP", "MF", "systematic_name", "number"), sep = "\\|\\|")
View(cleaned_data)
			    </code>
              </pre>
		    </section>
		    <section>
			  <h2>Autre possibilité</h2>
              <p>fonction <i>separate</i> de tidyr :</p>
			  <pre>
                <code class="R" data-trim>
library(dplyr)
library(tidyr)
cleaned_data <- separate(Brauer2008_DataSet1, NAME, c("name", "BP", "MF", "systematic_name", "number"), sep = "\\|\\|")
			    </code>
              </pre>
		    </section>
		  </section>

		  <!-- retrait des espaces -->
		    <section>
			  <h2>Élimination des espaces</h2>
			  <pre>
                <code class="R" data-trim>
head(cleaned_data$BP)
			    </code>
              </pre>
			  <pre>
## [1] " ER to Golgi transport "        " biological process unknown "  
## [3] " proteolysis and peptidolysis " " mRNA polyadenylylation* "     
## [5] " vesicle fusion* "              " biological process unknown "
              </pre>
		    </section>

		  <!-- retrait des espaces -->
		    <section>
			  <h2>Élimination des espaces</h2>
              <p>fonction trimws (<i>trim whitespace</i>) :</p>
			  <pre>
                <code class="R" data-trim>
cleaned_data <- Brauer2008_DataSet1 %>%
    separate(NAME, c("name", "BP", "MF", "systematic_name", "number"), sep = "\\|\\|") %>%
    mutate_each(funs(trimws), name:systematic_name)
			    </code>
              </pre>
              <pre>
[1] "ER to Golgi transport"        "biological process unknown"  
[3] "proteolysis and peptidolysis" "mRNA polyadenylylation*"     
[5] "vesicle fusion*"              "biological process unknown"
              </pre>
		    </section>

		  <!-- élimination de colonnes -->
		    <section>
			  <h2>Élimination de colonnes</h2>
              <p>fonction <i>select</i> de dplyr :</p>
			  <pre>
                <code class="R" data-trim>
cleaned_data <- Brauer2008_DataSet1 %>%
    separate(NAME, c("name", "BP", "MF", "systematic_name", "number"), sep = "\\|\\|") %>%
    mutate_each(funs(trimws), name:systematic_name) %>%
    select(-number, -GID, -YORF, -GWEIGHT)
			  </code>
            </pre>
            <pre>
Observations: 5537
Variables:
$ name            (chr) "SFB2", "", "QRI7", "CFT2", "SSO2", "PSP2", ...
$ BP              (chr) "ER to Golgi transport", "biological process...
$ MF              (chr) "molecular function unknown", "molecular fun...
$ systematic_name (chr) "YNL049C", "YNL095C", "YDL104C", "YLR115W", ...
$ G0.05           (dbl) -0.24, 0.28, -0.02, -0.33, 0.05, -0.69, -0.5...
$ G0.1            (dbl) -0.13, 0.13, -0.27, -0.41, 0.02, -0.03, -0.3...
$ G0.15           (dbl) -0.21, -0.40, -0.27, -0.24, 0.40, 0.23, -0.1...
...
              </pre>
		    </section>


		  <!-- Extraire les variables des noms de colonnes -->
		    <section>
			  <h2>Passage au format Wickham</h2>
              <p>objectif : une colonne par variable. Passage de 36
              colonnes à 3 (nutriment, taux de croissance, expression) avec la
              fonction <i>gather</i> de tidyr :</p>
			  <pre>
                <code class="R" data-trim>
cleaned_data <- Brauer2008_DataSet1 %>%
    separate(NAME, c("name", "BP", "MF", "systematic_name", "number"), sep = "\\|\\|") %>%
    mutate_each(funs(trimws), name:systematic_name) %>%
    select(-number, -GID, -YORF, -GWEIGHT) %>%
    gather(sample, expression, G0.05:U0.3)
			    </code>
              </pre>
		    </section>

		  <!-- Extraire les variables des noms de colonnes -->
		    <section>
			  <h2>Passage au format Wickham</h2>
              <p>objectif : une colonne par variable. Passage de 36
              colonnes à 3 (nutriment, taux de croissance, expression) avec la
              fonction <i>gather</i> de tidyr :</p>
              <pre>
Observations: 199332
Variables:
$ name            (chr) "SFB2", "", "QRI7", "CFT2", "SSO2", "PSP2", ...
$ BP              (chr) "ER to Golgi transport", "biological process...
$ MF              (chr) "molecular function unknown", "molecular fun...
$ systematic_name (chr) "YNL049C", "YNL095C", "YDL104C", "YLR115W", ...
$ sample          (fctr) G0.05, G0.05, G0.05, G0.05, G0.05, G0.05, G...
$ expression      (dbl) -0.24, 0.28, -0.02, -0.33, 0.05, -0.69, -0.5...
...
              </pre>
		    </section>

		  <!-- Extraire les variables des noms de colonnes -->
		    <section>
			  <h2>Séparation des variables nutriment et taux de croissance</h2>
              <p>fonction <i>separate</i> de tidyr :</p>
			  <pre>
                <code class="R" data-trim>
cleaned_data <- Brauer2008_DataSet1 %>%
    separate(NAME, c("name", "BP", "MF", "systematic_name", "number"), sep = "\\|\\|") %>%
    mutate_each(funs(trimws), name:systematic_name) %>%
    select(-number, -GID, -YORF, -GWEIGHT) %>%
    gather(sample, expression, G0.05:U0.3) %>%
    separate(sample, c("nutrient", "rate"), sep = 1, convert = TRUE)
			    </code>
			  </pre>
		    </section>

		  <!-- Extraire les variables des noms de colonnes -->
		    <section>
			  <h2>Séparation des variables nutriment et taux de croissance</h2>
              <p>fonction <i>separate</i> de tidyr :</p>
			  <pre>
Observations: 199332
Variables:
$ name            (chr) "SFB2", "", "QRI7", "CFT2", "SSO2", "PSP2", ...
$ BP              (chr) "ER to Golgi transport", "biological process...
$ MF              (chr) "molecular function unknown", "molecular fun...
$ systematic_name (chr) "YNL049C", "YNL095C", "YDL104C", "YLR115W", ...
$ nutrient        (fctr) G, G, G, G, G, G, G, G, G, G, G, G, G, G, G...
$ rate            (dbl) 0.05, 0.05, 0.05, 0.05, 0.05, 0.05, 0.05, 0....
$ expression      (dbl) -0.24, 0.28, -0.02, -0.33, 0.05, -0.69, -0.5...
			  </pre>
		    </section>

		  <!-- Selection d'un gène d'intérêt -->
		    <section>
			  <h2>Focus sur le gène LEU1</h2>
              <p>fonction <i>filter</i> de dplyr :</p>
			  <pre>
                <code class="R" data-trim>
cleaned_data %>%
  filter(name == "LEU1") %>%
  View()
			    </code>
			  </pre>
              <p>LEU1 est un gène participant à la synthèse de la
              leucine. On a 36 observations (6 conditions et 6 taux de
              croissance).</p>
		    </section>

		  <!-- Selection d'un gène d'intérêt -->
		  <section>
		    <section>
			  <h2>Visualisation avec ggplot2</h2>
              <p>fonction <i>filter</i> de dplyr :</p>
			  <pre>
                <code class="R" data-trim>
library(ggplot2)

cleaned_data %>%
  filter(name == "LEU1") %>%
  ggplot(aes(rate, expression, color = nutrient)) +
  geom_line()
			    </code>
			  </pre>
              <p>LEU1 est un gène participant à la synthèse de la
              leucine. On a 36 observations (6 conditions et 6 taux de
              croissance).</p>
		    </section>
		    <section>
			  <h2>ggplot2 ? gnéé ??</h2>
              <p>grammaire graphique :</br>données + géométrie + système de coordonnées = graphique</p>
			  <pre>
                <code class="R" data-trim>
ggplot(data = cleaned_data,
  aes(x = rate, y = expression, color = nutrient)) +
  geom_line()
			    </code>
			  </pre>
              <p>système de coordonnées cartésien par défaut (polaire,
              logarithmique, cartographique, ...)</p>
		    </section>
		  </section>

		  <!-- Selection d'un gène d'intérêt -->
		    <section>
			  <h2>Visualisation avec ggplot2</h2>
              <img
                 data-src="http://varianceexplained.org/figs/2015-11-19-tidy-genomics/unnamed-chunk-14-1.png"
                 >
		    </section>

		  <!-- Selection de plusieurs gènes d'intérêt -->
		    <section>
			  <h2>Focus sur les gènes LEU</h2>
			  <pre>
                <code class="R" data-trim>
cleaned_data %>%
  filter(BP == "leucine biosynthesis") %>%
  ggplot(aes(rate, expression, color = nutrient)) +
  geom_line() +
  facet_wrap(~name)
			    </code>
			  </pre>
		    </section>

		  <!-- Selection d'un gène d'intérêt -->
		    <section>
			  <h2>Visualisation avec ggplot2</h2>
              <img
                 data-src="http://varianceexplained.org/figs/2015-11-19-tidy-genomics/unnamed-chunk-15-1.png"
                 >
		    </section>

		  <!-- Selection de plusieurs gènes d'intérêt -->
		    <section>
			  <h2>Focus sur les gènes LEU</h2>
              <p>ajustement de droite (modèle linéaire, ou <i>lm</i>)</p>
			  <pre>
                <code class="R" data-trim>
cleaned_data %>%
  filter(BP == "leucine biosynthesis") %>%
  ggplot(aes(rate, expression, color = nutrient)) +
  geom_point() +
  geom_smooth(method = "lm", se = FALSE) +
  facet_wrap(~name)
			    </code>
			  </pre>
		    </section>

		  <!-- Selection d'un gène d'intérêt -->
		    <section>
			  <h2>Visualisation avec ggplot2</h2>
              <img
                 data-src="http://varianceexplained.org/figs/2015-11-19-tidy-genomics/unnamed-chunk-16-1.png"
                 >
		    </section>

		  <!-- Selection de métabolisme d'intérêt -->
		    <section>
			  <h2>Focus sur le métabolisme du soufre</h2>
			  <pre>
                <code class="R" data-trim>
cleaned_data %>%
  filter(BP == "sulfur metabolism") %>%
  ggplot(aes(rate, expression, color = nutrient)) +
  geom_point() +
  geom_smooth(method = "lm", se = FALSE) +
  facet_wrap(~name + systematic_name, scales = "free_y")
			    </code>
			  </pre>
		    </section>

		  <!-- Selection d'un gène d'intérêt -->
		    <section>
			  <h2>Visualisation avec ggplot2</h2>
              <img
                 data-src="http://varianceexplained.org/figs/2015-11-19-tidy-genomics/unnamed-chunk-17-1.png"
                 >
		    </section>

		  <!-- Toutes les commandes -->
		  <section>
			<h2>Script complet</h2>
			<pre>
              <code class="R" data-trim>
library(dplyr)
library(tidyr)
library(ggplot2)

Brauer2008_DataSet1 <- read.delim("/tmp/RtmpmIvzA5/data67167154fe1d")

cleaned_data <- Brauer2008_DataSet1 %>%
  separate(NAME, c("name", "BP", "MF", "systematic_name", "number"), sep = "\\|\\|") %>%
  mutate_each(funs(trimws), name:systematic_name) %>%
  select(-number, -GID, -YORF, -GWEIGHT) %>%
  gather(sample, expression, G0.05:U0.3) %>%
  separate(sample, c("nutrient", "rate"), sep = 1, convert = TRUE)

cleaned_data %>%
  filter(BP == "leucine biosynthesis") %>%
  ggplot(aes(rate, expression, color = nutrient)) +
  geom_point() +
  geom_smooth(method = "lm", se = FALSE) +
  facet_wrap(~name)
			  </code>
            </pre>
		  </section>

		  <!-- Analyses supplémentaires sur le même jeu de données -->
		  <section>
			<h2>Modeling gene expression with broom</h2>
			<h3>a case study in tidy analysis</h3>
            <a href="http://varianceexplained.org/r/tidy-genomics-broom/">
              <img width="200"
                   data-src="http://varianceexplained.org/images/david_robinson_picture2.jpg"
                   >
            </a>
            <p>par David Robinson</br><a href="http://varianceexplained.org/r/tidy-genomics-broom/">analyses supplémentaires sur le même jeu de données</a></p>
		  </section>

		  <!-- ggrepel -->
		  <section>
			<h2>ggrepel</h2>
			<br>
            <a href="https://cran.r-project.org/web/packages/ggrepel/vignettes/ggrepel.html">
			  <img width="720"
                   data-src="images/ggrepel.png"
                   alt="Repel overlapping text labels away from each other"
                   >
            </a>
		  </section>

		  <!-- ggbio -->
		  <section>
			<h2>ggbio</h2>
			<br>
            <a href="http://www.tengfei.name/ggbio/">
			  <img width="720"
                   data-src="https://raw.githubusercontent.com/tengfei/ggbio/gh-pages/images/interval.png"
                   alt="visualization of genomic data with ggbio"
                   >
            </a>
		  </section>

		  <!-- ggtree -->
		  <section>
			<h2>ggtree</h2>
			<br>
            <a href="http://bioconductor.org/packages/release/bioc/html/ggtree.html">
			  <img width="720"
                   data-src="images/ggtree.png"
                   alt="ggtree example with alignment"
                   >
            </a>
		  </section>
          
		  <!-- ggmap -->
		  <section>
			<h2>ggmap</h2>
			<br>
            <a href="http://chrisbeeley.net/?m=201207">
			  <img
                 data-src="http://chrisbeeley.net/wp-content/uploads/2012/07/map2.png"
                 alt="ggmap example"
                 >
            </a>
		  </section>

		  <!-- ggtern -->
		  <section>
			<h2>ggtern</h2>
			<br>
            <a href="http://www.ggtern.com">
			  <img
                 width="600"
                 data-src="http://www.ggtern.com/wp-content/uploads/2014/06/limit_region_off.png"
                 alt="ggtern example"
                 >
            </a>
		  </section>

		  <!-- gganimate -->
		  <section>
			<h2>gganimate</h2>
			<br>
            <a href="http://varianceexplained.org/files/bandwidth.html">
			  <img
                 width="500"
                 data-src="http://varianceexplained.org/images/density.gif"
                 alt="gganimate example"
                 >
            </a>
		  </section>

          <!-- ggnetwork -->
		  <section>
			<h2>ggnetwork</h2>
			<br>
            <a href="https://cran.r-project.org/web/packages/ggnetwork/vignettes/ggnetwork.html">
			  <img
                 width="600"
                 data-src="images/ggnetwork.png"
                 alt="ggnetwork example"
                 >
            </a>
		  </section>
          
		  <!-- Documentation -->
		  <section>
			<h2>Documentation</h2>
            <p>antisèches <a href="https://www.rstudio.com/resources/cheatsheets/">tidyr,
            dplyr, ggplot2, RStudio</a>,
            tutoriel <a href="http://r-statistics.co/ggplot2-Tutorial-With-R.html">ggplot2</a></p>
		  </section>

          <!-- section Etienne -->
				<section>
					<h3>Exemples avancés</h3>
					<p>
					
					
					<img width = "50%" data-src="images/owl.jpg" alt="owl" >
					
					</p>
				</section>

				<!-- Spatial and temporal datasets -->
				<section>
					<h3>Gerer des données temporelles/spatiales </h3>
					<p>
						Données d'abondances de cullicoides sur des piègages en France métropolitaine sur plusieurs années.
					</p>
							<img width = "50%" data-src="images/c_imicola_femelle_reference.jpg" alt="imicola" >

				</section>
			
							
				<section>
					<p>
					Format csv -> "comma separated value" 
					</p>
					<p>
						Les champs sont séparés par le charactère <strong>"|"</strong>
					</p>
							<img width = "100%" data-src="images/rawdataOcapi.png" alt="dataset" >

				</section>

				</section>
				<section>
					<h2>
						Import et manipulation des données 	
					</h2>

				</section>
			
	<section>
				<section>
					<pre><code class="R" data-noescape data-trim contenteditable>

data = read.table("uniqDataOcapi.csv",sep="|",header=T,na.strings = "NA",
fill=TRUE, strip.white=TRUE, blank.lines.skip = TRUE) 
data %>% tbl_df %>% View
				</code></pre>


							<img width = "100%" data-src="images/Ocapi.png" alt="dataset" >


				</section>




				<section>

					<pre><code class="R" data-noescape data-trim contenteditable>

td = data %>% tbl_df %>% na.omit %>% mutate(.,lat = as.numeric(LAT_IDENTIF))
%>% mutate(.,long = as.numeric(LONG_IDENTIF)) %>%  
<mark> mutate(.,date =as.Date(DATEDEBUT,"%m/%d/%Y")) %>% </mark>
 mutate(.,datesp = date) %>%
separate(.,datesp,c("year","month","day")) %>%
select(.,ID_SITE,long,lat,date,year,month,ESPECE,NBINDIV,NBMALES,NBFEMELLES)

				</code></pre>

					<pre><code class="R" data-noescape data-trim contenteditable>
> td
Source: local data frame [54,634 x 10]

   ID_SITE  long   lat       date  year month             ESPECE NBINDIV
    (fctr) (dbl) (dbl)     (<mark>date</mark>) (chr) (chr)             (fctr)   (int)
1    01PL1  5.79 46.06 2009-10-12  2009    10     Culicoides sp.       0
2    01PL1  5.79 46.06 2009-10-19  2009    10     Culicoides sp.       0
3    01PL1  5.79 46.06 2009-10-26  2009    10 obsoletus/scoticus       4
4    01PL1  5.79 46.06 2009-10-05  2009    10          lupicaris      11
5    01PL1  5.79 46.06 2009-10-05  2009    10 obsoletus/scoticus     226
6    01PL1  5.79 46.06 2009-10-05  2009    10     obsoletus s.st       1
7    01PL1  5.79 46.06 2009-10-05  2009    10          pulicaris      14
8    01PL1  5.79 46.06 2009-11-02  2009    11     Culicoides sp.       0
9    01PL1  5.79 46.06 2009-11-23  2009    11 obsoletus/scoticus       2
10   01PL1  5.79 46.06 2009-11-30  2009    11     Culicoides sp.       0
..     ...   ...   ...        ...   ...   ...                ...     ...
Variables not shown: NBMALES (int), NBFEMELLES (int)



					</code></pre>

				</section>
				
				<section>
					<h3> Manipulation des dates </h3>
					
					<p>
					grace au fonction as.Date() et format()
					</p>
						<img width = "100%" data-src="images/format_date.png" alt="date" >

				</section>

		</section>

		<section>
				<section>
					<h3> Affichage des données avec ggplot</h3>

				</section> 


				<section>
					<pre><code class="R" data-noescape data-trim contenteditable>

td %>%  ggplot() + geom_boxplot(aes(group = months ,x=date,y= NBINDIV))  +
scale_y_log10() +theme_bw() 
					</code></pre>

					<img width = "100%" data-src="images/culicoide/p1.png" alt="p1" >
				</section>

				<section>
					<pre><code class="R" data-noescape data-trim contenteditable>
										
td %>% ggplot() + geom_boxplot(aes(group = months ,x=date,y= NBINDIV))  +
scale_y_log10()  + theme_bw() + scale_x_date(breaks = "6 month") 

					</code></pre>

					<img width = "100%" data-src="images/culicoide/p2.png" alt="p1" >
				</section>

	
				<section>
					<pre><code class="R" data-noescape data-trim contenteditable>
										
td %>%   ggplot() + geom_boxplot(aes(group = months ,x=date,y= NBINDIV))  +
scale_y_log10()  + theme_bw() + scale_x_date(breaks = "1 month", labels =
date_format("%b\n%Y"), limits = as.Date(c("2009-01-01","2010-01-01"))) 

					</code></pre>

					<img width = "100%" data-src="images/culicoide/p3.png" alt="p1" >
				</section>
				<section>

					<pre><code class="R" data-noescape data-trim contenteditable>
> td
Source: local data frame [54,634 x 10]

   ID_SITE  long   lat       date  year <mark>month</mark>             ESPECE NBINDIV
    (fctr) (dbl) (dbl)     (date) (chr) (chr)             (fctr)   (int)
1    01PL1  5.79 46.06 2009-10-12  2009    10     Culicoides sp.       0
2    01PL1  5.79 46.06 2009-10-19  2009    10     Culicoides sp.       0
3    01PL1  5.79 46.06 2009-10-26  2009    10 obsoletus/scoticus       4
4    01PL1  5.79 46.06 2009-10-05  2009    10          lupicaris      11
5    01PL1  5.79 46.06 2009-10-05  2009    10 obsoletus/scoticus     226
6    01PL1  5.79 46.06 2009-10-05  2009    10     obsoletus s.st       1
7    01PL1  5.79 46.06 2009-10-05  2009    10          pulicaris      14
8    01PL1  5.79 46.06 2009-11-02  2009    11     Culicoides sp.       0
9    01PL1  5.79 46.06 2009-11-23  2009    11 obsoletus/scoticus       2
10   01PL1  5.79 46.06 2009-11-30  2009    11     Culicoides sp.       0
..     ...   ...   ...        ...   ...   ...                ...     ...
Variables not shown: NBMALES (int), NBFEMELLES (int)



					</code></pre>

				</section>
				

	
	
	

				<section>
					<pre><code class="R" data-noescape data-trim contenteditable>
td %>%  ggplot() + geom_boxplot(aes(x=month,y= NBINDIV ,fill=year)) +
scale_y_log10() + scale_fill_brewer(palette = "Set1") + theme_bw()

					</code></pre>

					<img width = "100%" data-src="images/culicoide/p4.png" alt="p1" >
				</section>

	
					<section>
						<h3>Recherche des espèces les plus représentées</h3>
					<pre><code class="R" data-noescape data-trim contenteditable>

td %>% group_by(ESPECE) %>% summarize(mean(NBINDIV)) %>% top_n(.,10)
					</code></pre>
					<pre><code class="R" data-noescape data-trim contenteditable>
Selecting by mean(NBINDIV)
Source: local data frame [10 x 2]

               ESPECE mean(NBINDIV)
               (fctr)         (dbl)
1          abchazicus     124.33333
2            deltus ?     216.48458
3             dewulfi     181.55324
4         furcillatus      81.65903
5          grisescens     399.00000
6             imicola     860.28247
7           newsteadi      88.92803
8  obsoletus/scoticus     433.05892
9              saevus     111.71429
10           tauricus     780.00000

					</code></pre>
				</section>


				<section>
					<pre><code class="R" data-noescape data-trim contenteditable>

td %>% filter(ESPECE == "imicola")  %>% filter(year %in%
c("2009","2010","2011"))  %>% group_by(year) %>%  ggplot() +
geom_boxplot(aes(x=month,y= NBINDIV ,fill=year)) + scale_y_log10() +
scale_fill_brewer(palette = "Set2") + theme_bw() 
					</code></pre>

					<img width = "100%" data-src="images/culicoide/p5.png" alt="p1" >
				</section>


				<section>
					<pre><code class="R" data-noescape data-trim contenteditable>

tdfilt = td %>% filter(ESPECE %in% c("imicola","dewulfi","chiopterus"))  %>%
filter(year %in% c("2009","2010","2011"))					
					
tdfilt %>% group_by(year) %>%  ggplot() + geom_boxplot(aes(x=month,y= NBINDIV
,fill=year)) + scale_y_log10() + scale_fill_brewer(palette = "Set2") +
theme_bw() + facet_grid(ESPECE ~ .)
					</code></pre>

					<img width = "100%" data-src="images/culicoide/p6.png" alt="p1" >
				</section>


				<section>
					<pre><code class="R" data-noescape data-trim contenteditable>

tdfilt %>% group_by(year) %>%  ggplot() + geom_boxplot(aes(x=month,y= NBINDIV
,fill=year)) + scale_y_log10() + scale_fill_brewer(palette = "Set2") +
theme_bw() + facet_grid(year ~ ESPECE)
					</code></pre>

					<img width = "100%" data-src="images/culicoide/p7_1.png" alt="p1" >
				</section>


				<section>
					<pre><code class="R" data-noescape data-trim contenteditable>

tdfilt %>% group_by(year) %>%  ggplot() + geom_violin(aes(x=month,y= NBINDIV
,fill=year),scale = "area", trim = FALSE  ) + scale_y_log10() +
scale_fill_brewer(palette = "Set2") +
theme_bw() + facet_grid(year ~ ESPECE)  </code></pre>

					<img width = "100%" data-src="images/culicoide/p7.png" alt="p1" >
				</section>


				<section>
					<pre><code class="R" data-noescape data-trim contenteditable>

tdfilt %>% group_by(year) %>%  ggplot() + geom_jitter(aes(x=month,y= NBFEMELLES
,color=year) ) + scale_y_log10() + scale_fill_brewer(palette = "Set2") +
theme_bw() + facet_grid(year ~ ESPECE)
					</code></pre>

					<img width = "100%" data-src="images/culicoide/p8.png" alt="p1" >
				</section>

				<section>
					<h3>Pour changer le format d'affichage des mois "a la main"</h3>

					<pre><code class="R" data-noescape data-trim contenteditable>

monthformat = function(x){
return(format(as.Date(paste("1",x,sep="/"),format="%d/%m"),"%b"))}

					</code></pre>

					<pre><code class="R" data-noescape data-trim contenteditable>

> monthformat("10")
[1] "oct."

					</code></pre>

				</section>


				<section>
					<pre><code class="R" data-noescape data-trim contenteditable>

tdfilt %>% group_by(year) %>%  ggplot() + geom_boxplot(aes(x=month,y=
NBFEMELLES ,fill=year)) + scale_y_log10(labels = comma) +
scale_x_discrete(labels = monthformat)
+  scale_fill_brewer(palette = "Set2",name="année") + theme_bw() +
facet_grid(year ~ ESPECE) +
theme(axis.text.x = element_text(angle = 45, hjust = 1)) +  xlab(NULL) +
ylab("Captures /  Pièges / Nuit") +
theme(text = element_text(family="Times", size = 14))
				       	</code></pre>

					<img width = "100%" data-src="images/culicoide/p9.png" alt="p1" >
				</section>


				<section>
					<pre><code class="R" data-noescape data-trim contenteditable>

tdfilt %>% group_by(year) %>%  ggplot() + geom_boxplot(aes(x=month,y=
NBFEMELLES )) + scale_y_log10(labels = comma) + scale_x_discrete(labels =
monthformat) + theme_bw() +
facet_grid(year ~ ESPECE) + theme(axis.text.x = element_text(angle = 45, hjust
= 1)) +  xlab(NULL) + ylab("Captures /
Pièges / Nuit") + theme(text = element_text(family="Times", size = 14))
					</code></pre>

					<img width = "100%" data-src="images/culicoide/p10.png" alt="p1" >
				</section>


				<section>
					<pre><code class="R" data-noescape data-trim contenteditable>

tdfilt %>% regroup(list("year","month")) %>% mutate(Sexratio =
mean(NBFEMELLES/NBINDIV)) %>% group_by(year) %>%  ggplot() +
geom_boxplot(aes(x=month,y= NBINDIV
,fill = Sexratio)) + scale_y_log10(labels = comma) + scale_x_discrete(labels =
monthformat) + theme_bw() +
facet_grid(year ~ ESPECE) + theme(axis.text.x = element_text(angle = 45, hjust
= 1)) +  xlab(NULL) + ylab("Captures /
Pièges / Nuit") + theme(text = element_text(family="Times", size = 14))
					</code></pre>

					<img width = "90%" data-src="images/culicoide/p11.png" alt="p1" >
				</section>

		</section>
				
		
				<section>
				<h1>Données spatiales !</h1>
				</section>	
	
		<section>
				<section>
				<h3>ggmap</h3>
				<p> Construit sur la même grammaire que ggplot, relativement accessible
				</p>
				</section>	

				<section>
				<h3>On recupère la carte sur laquelle les données sont projetées: </h3>
					<pre><code class="R" data-noescape data-trim contenteditable>

?get_map

Grab a map.

Description:

‘get_map’ is a smart wrapper that queries the Google Maps,
OpenStreetMap, Stamen Maps or Naver Map servers for a map.

Usage:

get_map(location = c(lon = -95.3632715, lat = 29.7632836), zoom = "auto",
scale = "auto", maptype = c("terrain", "terrain-background", "satellite",
"roadmap", "hybrid", "toner", "watercolor", "terrain-labels", "terrain-lines",
"toner-2010", "toner-2011", "toner-background", "toner-hybrid",
"toner-labels", "toner-lines", "toner-lite"), source = c("google", "osm",
"stamen", "cloudmade"), force = ifelse(source == "google", TRUE, TRUE),
messaging = FALSE, urlonly = FALSE, filename = "ggmapTemp",
crop = TRUE, color = c("color", "bw"), language = "en-EN", api_key)

					</code></pre>
				</section>

				<section>
				<h3>On recupère la carte sur laquelle les données sont projetées: </h3>

					<pre><code class="R" data-noescape data-trim contenteditable>

France = get_map(location = "France",zoom = 5,  source = "google",maptype =
"satellite",color = "bw") 
ggmap(France)

					</code></pre>
					<img width = "90%" data-src="images/culicoide/p12.png" alt="p1" >
				</section>
				
				<section>

					<pre><code class="R" data-noescape data-trim contenteditable>



ggmap(France) + geom_point(data = tdfilt %>%
regroup(list("year","ID_SITE","ESPECE")) %>% mutate(avg = log(mean(NBFEMELLES))), aes(x = long,
y=lat, size = avg)) + facet_grid(year ~ ESPECE) + scale_x_continuous(limits =
c(-6.5,9.5)) +
scale_y_continuous(limits = c(41,51) ) + scale_size_continuous(name="Moyenne
Captures/femelles/nuit",breaks=c(0,1,2,3,4,5),labels
=c(1,10,100,1000,10000,100000)) 
					
					</code></pre>
					<img width = "90%" data-src="images/culicoide/p13.png" alt="p1" >
				</section>
				<section>
			<h3>Une carte par mois ?</h3>


					<pre><code class="R" data-noescape data-trim contenteditable>
months = unique(tdfilt$month)
>months
[1] "04" "07" "08" "10" "11" "05" "06" "09" "03" "12" "01" "02"
>months[sort.list(months)]
[1] "01" "02" "03" "04" "05" "06" "07" "08" "09" "10" "11" "12"

					</code></pre>
				</section>


				<section>

					<pre><code class="R" data-noescape data-trim contenteditable>
saveGIF({
	for(i in months[sort.list(months)])
		{ print(ggmap(France) +
		geom_point(data = tdfilt %>% filter(month == i) %>% regroup(list("year","ID_SITE","ESPECE")) %>% mutate(avg = log(mean(NBFEMELLES))), aes(x = long, y=lat, size = avg)) + facet_grid(year ~ ESPECE) + scale_x_continuous(limits = c(-6.5,9.5)) + scale_y_continuous(limits= c(41,51) ) +  ggtitle(monthformat(i)) + scale_size_continuous(name="Moyenne Captures/femelles/nuit", limits=c(0,6),breaks  =c(0,1,2,3,4,5),labels=c(1,10,100,1000,10000,100000)))}
	}, interval = 1,movie.name="Culicoids.gif")


					</code></pre>


					<img width = "60%" data-src="images/culicoide/Culicoids.gif" alt="p0" >
				</section>
		</section>
		
		<section>

				<section>
				<h3>
					Données "PopPhyl"
				</h3>

				<img width = "100%" data-src="images/popphyl/pophyl_logo.jpg" alt="p0" >
				</section>

					
				<section>

					<pre><code class="R" data-noescape data-trim contenteditable>
dp = read.table("data_and_script_part2/popphyl_data.csv",sep=",",header = T)
dp %>% tbl_df %>% View	
					</code></pre>
					<img width = "100%" data-src="images/popphyl/rawpop.png" alt="p1" >
				</section>
					
				<section>

					<pre><code class="R" data-noescape data-trim contenteditable>
					
dp %>% ggplot() + geom_point(aes(x=adult_size,y=piS))
					
					</code></pre>

					<img width = "100%" data-src="images/popphyl/p1.png" alt="p1" >
				</section>

					
				<section>

					<pre><code class="R" data-noescape data-trim contenteditable>
dp %>% ggplot() + geom_point(aes(x=adult_size,y=piS*100)) +
scale_x_log10(limits = c(0.1,1000), breaks = c(1,10,100)) + scale_y_log10()
					</code></pre>
					<img width = "100%" data-src="images/popphyl/p2.png" alt="p1" >
				</section>

					
				<section>

					<pre><code class="R" data-noescape data-trim contenteditable>
dp %>% ggplot(aes(x = adult_size, y = piS)) + geom_point(size = 3,aes(shape =
Taxo_for_fig  )) + scale_x_log10(limits = c(0.1,100), breaks = c(1,10,100)) +
scale_y_log10() 
					
					</code></pre>
					<img width = "100%" data-src="images/popphyl/p3.png" alt="p1" >
				</section>

					
				<section>

					<pre><code class="R" data-noescape data-trim contenteditable>
baseplot = dp %>%  mutate(parental_invest = propagule_size / adult_size) %>%
na.omit  %>% ggplot(aes(x=adult_size,y =piS)) + geom_point(size =3, aes(color =
parental_invest, shape = Taxo_for_fig  )) + scale_x_log10(limits = c(0.1,100),
breaks = c(1,10,100)) + scale_y_log10() 
baseplot + scale_color_gradientn(colours = rainbow(7) , trans = "log", limits =
c(0.0005,1), values =c (0.001,0.01,0.1,1), breaks = c(0.001,0.01,0.1,1), labels
= comma)  + theme_bw()

					</code></pre>
					<img width = "100%" data-src="images/popphyl/p4.png" alt="p1" >
				</section>

					
				<section>

					<pre><code class="R" data-noescape data-trim contenteditable>
baseplot + scale_color_gradientn(colours = rainbow(7) , trans = "log", limits =
c(0.0005,1), values =c (0.001,0.01,0.1,1), breaks = c(0.001,0.01,0.1,1), labels
= comma)  + theme_bw() +
geom_smooth(method = "lm", se = TRUE, size = 0.7, color = "black", formula = y
					~ x) </code></pre>
					<img width = "100%" data-src="images/popphyl/p5.png" alt="p1" >
				</section>

					
				<section>
					<h3>Ajouter une equation ...</h3>
					
					<a href="http://stackoverflow.com/questions/7549694/ggplot2-adding-regression-line-equation-and-r2-on-graph">Ou trouver cette info ?</a>
					<pre><code class="R" data-noescape data-trim contenteditable>
	require(devtools)
source_gist("524eade46135f6348140")

				
					</code></pre>
				</section>

					
				<section>

					<pre><code class="R" data-noescape data-trim contenteditable>
baseplot + scale_color_gradientn(colours = rainbow(7) , trans = "log", limits =
c(0.0005,1), values =c (0.001,0.01,0.1,1), breaks = c(0.001,0.01,0.1,1), labels
= comma)  + theme_bw() +
geom_smooth(method = "lm", se = TRUE, size = 0.7, color = "black", formula = y
~ x) + stat_smooth_func(geom="text",
method = "lm", hjust= 0, parse =TRUE ) </code></pre>
					<img width = "100%" data-src="images/popphyl/p6.png" alt="p1" >
				</section>

					
				<section>

					<pre><code class="R" data-noescape data-trim contenteditable>
dp = read.table("data_and_script_part2/popphyl_data.csv",sep=",",header = T)
dp %>%  mutate(parental_invest = propagule_size / adult_size) %>% na.omit  %>%
ggplot(aes(x=adult_size,y =piS)) + geom_point(size =3, aes(color =
parental_invest, shape = Taxo_for_fig
)) + scale_x_log10(limits = c(0.1,100), breaks = c(1,10,100)) + scale_y_log10()
+ scale_color_gradientn(name =
"Parental investment",colours = rainbow(7) , trans = "log", limits =
c(0.0005,1), values =c
(0.001,0.01,0.1,1), breaks = c(0.001,0.01,0.1,1), labels = comma) +
scale_shape(name = "Taxonomic group") +
theme_bw() + geom_smooth(method = "lm", se = TRUE, size = 0.7, color = "black",
formula = y ~ x) +
stat_smooth_func(geom="text", method = "lm", hjust= 0, parse =TRUE ) +
xlab("Adult size (cm)") +
ylab(expression(paste(pi,"s"))) +  theme(text=
element_text(family="Bookman",size=14))
					</code></pre>
				</section>	
				<section>	
					
					<img width = "100%" data-src="images/popphyl/p7.png" alt="p1" >
				</section>

				<section>
					<h3>Graph interactif avec plotly </h3>
				
					<pre><code class="R" data-noescape data-trim contenteditable>
			
plot_ly(dp %>%   mutate(parental_invest = propagule_size / adult_size) %>%
na.omit , x = adult_size, y = piS, text = paste("Species: ", Species,"\n
Family: ",Taxo_for_fig), color =
log(parental_invest), mode = "markers")  %>% layout(xaxis = list(title = "Adult
size " , type = "log"))  %>%
layout(yaxis = list(title = "Pi_s", type = "log"))  </code></pre>
				</section>

				<section>
					<iframe src = "interpop.html" width ="800" height= "500" frameborder = 0>
					</iframe>
				</section>



		</section>



					<section id="Conclusions">
						<h2>Conclusions</h2>
						<p>Tout est possible</p>
						<p class="fragment">Rien ne va marcher</p>
						<p class="fragment">Mais quelqu'un a déjà essayé</p>

					</section>


	
	
	
	
			<!-- Conclusion -->
				<section style="text-align: left;">
				  <h1>THE END</h1>
				  <p>
					- <a href="http://slides.com">Try the online editor</a> <br>
					- <a href="https://github.com/hakimel/reveal.js">Source code &amp; documentation</a>
				  </p>
				</section>

			</div>
		</div>

		<script src="lib/js/head.min.js"></script>
		<script src="js/reveal.js"></script>

		<script>

			// Full list of configuration options available at:
			// https://github.com/hakimel/reveal.js#configuration
			Reveal.initialize({
				controls: true,
				progress: true,
				history: true,
				center: true,

				transition: 'slide', // none/fade/slide/convex/concave/zoom

				// Optional reveal.js plugins
				dependencies: [
					{ src: 'lib/js/classList.js', condition: function() { return !document.body.classList; } },
					{ src: 'plugin/markdown/marked.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
					{ src: 'plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
					{ src: 'plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
					{ src: 'plugin/zoom-js/zoom.js', async: true },
					{ src: 'plugin/notes/notes.js', async: true }
				]
			});

		</script>

	</body>
</html>
